src/main/webapp/app/exercises/text/participate/text-editor.component.ts

```diff
@@ -131,7 +178,12 @@ export class TextEditorComponent implements OnInit, OnDestroy, ComponentCanDeact
         }
     }
 
-    private updateParticipation(participation: StudentParticipation) {
+    /**
+     * Updates the participation, the submission selected can be chosen through submissionId, default undefined means latest
+     * @param participation The participation data
+     * @param submissionId The id of the submission of choice. undefined value defaults to the latest submission
+     */
+    private updateParticipation(participation: StudentParticipation, submissionId: number | undefined = undefined) {
         this.participation = participation;
         this.textExercise = this.participation.exercise as TextExercise;
```

Comment:
"Cannot read properties of undefined (reading 'exercise')"

Reply:
I think this happens after I vote "How useful is this feedback to you?" and refresh on an older result